react/features/toolbox/components/AbstractToolbarButton.js

```diff
@@ -1,61 +1,66 @@
-import PropTypes from 'prop-types';
+// @flow
+
 import React, { Component } from 'react';
 
 /**
- * Abstract (base) class for a button in {@link Toolbar}.
- *
- * @abstract
+ * The type of the React {@link Component} props of
+ * {@link AbstractToolbarButton}.
  */
-export default class AbstractToolbarButton extends Component {
+export type Props = {
+
     /**
-     * AbstractToolbarButton component's property types.
-     *
-     * @static
+     * A succinct description of what the button does. Used by accessibility
+     * tools and torture tests.
+     */
+    accessibilityLabel?: string,
+
+    /**
+     * The name of the Icon of this {@code AbstractToolbarButton}.
+     */
+    iconName?: string,
+
+    /**
+     * The style of the Icon of this {@code AbstractToolbarButton}.
+     */
+    iconStyle?: Object,
+
+    /**
+     * On click handler.
      */
-    static propTypes = {
-        /**
-         * The name of the Icon of this {@code AbstractToolbarButton}.
-         */
-        iconName: PropTypes.string,
-
-        /**
-         * The style of the Icon of this {@code AbstractToolbarButton}.
-         */
-        iconStyle: PropTypes.object,
-
-        /**
-         * On click handler.
-         */
-        onClick: PropTypes.func,
-
-        /**
-         * {@code AbstractToolbarButton} styles.
-         */
-        style:
-            PropTypes.oneOfType([
-                PropTypes.array,
-                PropTypes.object
-            ]),
-
-        /**
-         * The color underlaying the button.
-         */
-        underlayColor: PropTypes.any
-    };
+    onClick: Function,
 
+    /**
+     * {@code AbstractToolbarButton} styles.
+     */
+    style?: Object,
+
+    /**
+     * The color underlaying the button.
+     */
+    underlayColor?: ?any
+};
+
+/**
+ * Abstract (base) class for a button in {@link Toolbar}.
+ *
+ * @abstract
+ */
+export default class AbstractToolbarButton<P : Props> extends Component<P> {
```

Comment:
Wy do you need to alias P to Props? Can't we just use Props like elsewhere?

Reply:
The answer to every question you've asked is: I don't know what I'm doing and I did whatever I could to make flow stop complaining. This specifically is the pattern I saw being used for AbstractDialog and I couldn't find another combination of flow syntax to let the Props be inherited without doing the `P : Props` thing.